src/App.tsx

```diff
@@ -1,111 +1,321 @@
-import { useState } from 'react';
-import axios from 'axios';
+import { useState, useEffect } from 'react';
+import axios, { AxiosResponse } from 'axios';
 import Button from '@mui/material/Button';
 import SendIcon from '@mui/icons-material/Send';
 import Card from '@mui/material/Card';
 import CardContent from '@mui/material/CardContent';
 import Checkbox from '@mui/material/Checkbox';
+import FormControlLabel from '@mui/material/FormControlLabel';
 import ButtonGroup from '@mui/material/ButtonGroup';
-import Grid from '@mui/material/Grid';
-import Stack from '@mui/material/Stack';
 import Typography from '@mui/material/Typography';
 import Tooltip from '@mui/material/Tooltip';
 import Zoom from '@mui/material/Zoom';
-import example from './example.json';
+import TextField from '@mui/material/TextField';
+import Autocomplete from '@mui/material/Autocomplete';
+import { queryURL, attributesURL, sexes, modalities } from './utils/constants';
 
 import './App.css';
 
-const exampleResult: Result[] = example as Result[];
-
-function ResultCard({nodeName, datasetName, datasetTotalSubjects, numMatchingSubjects, imageModals} : {nodeName: string, datasetName: string, datasetTotalSubjects: number, numMatchingSubjects: number, imageModals: string[]}) {
-  const modalities: {[key: string]: {name: string, style: string}} = {
-    'http://purl.org/nidash/nidm#ArterialSpinLabeling': {
-      name: 'ASL',
-      style: 'bg-zinc-800 text-white',
-    },
-    'http://purl.org/nidash/nidm#DiffusionWeighted': {
-      name: 'DWI',
-      style: 'bg-red-700 text-white',
-    },
-    'http://purl.org/nidash/nidm#EEG':
-    {
-      name: 'EEG',
-      style: 'bg-rose-300 text-white',
-    },
-    'http://purl.org/nidash/nidm#FlowWeighted':
-    {
-      name: 'Flow',
-      style: 'bg-sky-700 text-white',
-    },
-    'http://purl.org/nidash/nidm#T1Weighted': {
-      name: 'T1',
-      style: 'bg-yellow-500 text-white',
-    },
-    'http://purl.org/nidash/nidm#T2Weighted': {
-      name: 'T2',
-      style: 'bg-green-600 text-white',
-    },
-  };
+const modalityOptions: object = modalities;
+
+function CategoricalField({
+  label,
+  options,
+  onFieldChange,
+}: {
+  label: string;
+  options: InputOption[];
+  onFieldChange: (fieldLabel: string, value: string | null) => void;
+}) {
+  return (
+    <Autocomplete
+      disablePortal
+      options={options}
+      renderInput={(params) => (
+        <TextField
+          // eslint-disable-next-line react/jsx-props-no-spreading
+          {...params}
+          InputLabelProps={params.InputLabelProps}
+          label={label}
+          className="w-full"
+        />
+      )}
+      onChange={(_, value) => onFieldChange(label, value?.id ?? null)}
```

Comment:
aha, took me a second to get that `??` here means that if the left is either `undefined` or `null`, then the right will always be just `null`. good to know